app/src/main/java/com/example/concordia_campus_guide/Helper/RoutesHelpers/DirectionsApiDataParser.java

```diff
@@ -51,6 +56,59 @@ protected DirectionsResult doInBackground(DirectionsApiDataRetrieval... obj) {
      */
     @Override
     protected void onPostExecute(DirectionsResult result) {
-        dataRetrieval.caller.directionsApiCallBack(result);
+        dataRetrieval.caller.directionsApiCallBack(result, extractRelevantInfoFromDirectionsResultObj(result));
+    }
+
+    public List<Route> extractRelevantInfoFromDirectionsResultObj(DirectionsResult result) {
```

Comment:
should comment this method

Reply:
could rename to extractListOfRoutesFromDirectionsAPIResponseObject, would avoid need for comment